File: RR.Frontend/src/DTOs/Product.ts
export class Product {
  id?: number;
  userId?: number;
  groupId?: number;
  isPublic: boolean;
  name: string;
  description?: string;
  quantity?: number;
  measurement?: string;
  imageUrl?: string;

  constructor() {
    this.isPublic = true;
    this.name = '';
    this.description = '';
    this.quantity = undefined;
    this.measurement = '';
    this.imageUrl = '';
  }

  static from(obj: any): Product {
    const p = Object.assign(new Product(), obj);
    return p;
  }

  isEmpty(): boolean {
    return (!this.name || this.name.trim() === '') &&
      (!this.description || this.description.trim() === '') &&
      (!this.quantity || this.quantity === 0) &&
      (!this.measurement || this.measurement === '') &&
      (!this.imageUrl || this.imageUrl.trim() === '');
  }
}

File: RR.Common/Models/Product.cs
namespace RR.Common.Models;

public class Product : IOwnable
{
    public string? Name { get; set; }
    public int? Id { get; set; }
    public int? GroupId { get; set; }
    public bool? IsPublic { get; set; }
    public string? Description { get; set; }
    public float? Quantity { get; set; }
    public Measurement? Measurement { get; set; }
    public string? ImageUrl { get; set; }
    public string ToIdAndNameString() => $"{{ Id: {Id}, Name: {Name} }}";
}
